### ISSUE TITLE
linter: --type-check-only reports regular suppressions as unused

### ISSUE THREAD

**Issue by hyoban** [oxc-project/oxc#24459]
### What version of Oxlint are you using?

`1.73.0` built from `main` at `9c4e24a365989f034b3c44f8390e67ee14b5c43f`.

### What command did you run?

```sh
oxlint --type-check-only index.ts
```

### What does your `.oxlintrc.json` config file look like?

```json
{
  "rules": {
    "no-console": "error"
  }
}
```

The suppression file contains the still-valid regular lint violation:

```json
{
  "index.ts": {
    "no-console": {
      "count": 1
    }
  }
}
```

`index.ts` contains:

```ts
console.log("still suppressed");
```

### What happened?

Regular single-file lint succeeds because the `no-console` violation still matches the suppression baseline:

```sh
oxlint index.ts
# Found 0 warnings and 0 errors.
```

However, `oxlint --type-check-only index.ts` exits with `LintUnprunedSuppressions` and reports that suppressions no longer occur.

In type-check-only mode, regular lint rules are intentionally skipped. The suppression runtime map therefore marks `index.ts` as seen but has no `no-console` count. The unconditional suppression finalization interprets that missing rule as a zero count instead of a rule that was not run.

### Expected behavior

`--type-check-only` should not validate or update the bulk suppression baseline because it only has a partial rule result. It should still report TypeScript compiler diagnostics normally.

`--suppress-all` and `--prune-suppressions` should be rejected when combined with `--type-check-only`, since a partial rule result cannot safely update the baseline.

### AI usage

I used OpenAI Codex to help investigate, minimize, and prepare this report. I reviewed the reproduction and take responsibility for this submission.
